### PR TITLE
Updated the AutoDiscoveryServiceImpl save behavior, and add unit tests

### PR DESCRIPTION
**Pull request by danielyhuang**
When updating an existing AutoDiscovery record, only update the status, not any other fields.
Created unit tests for create/update happy paths, and exception handling when updating existing records.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -123,7 +123,7 @@
                     <artifactId>mockito-all</artifactId>
                 </exclusion>
             </exclusions>
-            <version>3.1.4-SNAPSHOT</version>
+            <version>3.1.3</version>
         </dependency>
```
Comment (benj58xu) [inline pom.xml]: Is this change intentional?

File: src/main/java/com/capitalone/dashboard/service/AutoDiscoveryServiceImpl.java [hunk 2]
```diff
@@ -49,4 +56,38 @@ private AutoDiscovery requestToAutoiscovery(AutoDiscoveryRemoteRequest request)
...
+
+    /**
+     * Update the AutoDiscovery Entries' status from source to target.
+     * @param source
+     * @param target
+     */
+    private void updateEntryStatus(List<AutoDiscoveredEntry> source, List<AutoDiscoveredEntry> target) {
+        for (AutoDiscoveredEntry srcEntry : source) {
+            target.stream().forEach(entry -> {
+                if(entry.getOptions().equals(srcEntry.getOptions())){
```
Comment (nireeshT) [inline src/main/java/com/capitalone/dashboard/service/AutoDiscoveryServiceImpl.java]: @danielyhuang  does this checks references or values

File: pom.xml (hunk 1 above)
Comment (rvema) [reply pom.xml]: @benj58xu Yes for now, For api we need to give release versions, I am working on release of hygieia-core once that is done this version will get changed.

File: src/main/java/com/capitalone/dashboard/service/AutoDiscoveryServiceImpl.java (hunk 2 above)
Comment (danielyhuang) [reply src/main/java/com/capitalone/dashboard/service/AutoDiscoveryServiceImpl.java]: Yes, it compares key-value pair if both are String type, which is our case. If the value is a List, then we need special handling.

File: pom.xml (hunk 1 above)
Comment (rvema) [reply pom.xml]: @daniel14y70huang Please set the version property of `${com.capitalone.dashboard.core.version}` to 3.1.5

**Comment by danielyhuang** [comment]
close.
